### PR TITLE
Enable Azure task in the remote store manifest example DAG

### PR DESCRIPTION
**Pull request by pankajkoti**
During the work on PR https://github.com/astronomer/astronomer-cosmos/pull/1297, an issue arose where the Azure remote manifest task began failing after installing providers and packages with constraints. To allow other tests, which were running successfully, to proceed, the task was temporarily disabled. Upon reviewing the GitHub Actions logs from previous successful runs, it was [observed](https://github.com/astronomer/astronomer-cosmos/actions/runs/11573670971/job/32216315282#step:6:250) that the Azure provider version installed was 10.5.1. However, after the refactoring introduced in the PR, the failing actions [showed](https://github.com/astronomer/astronomer-cosmos/actions/runs/11911545582/job/33193301710#step:6:474) azure provider==8.4.0 being installed with the constraints file.

To investigate, I tested locally to identify a working version. While the task failed with 8.4.0, it succeeded with 8.5.0. Analyzing the failure [logs](https://github.com/astronomer/astronomer-cosmos/actions/runs/11911545582/job/33193301710#step:7:467) and reviewing the Azure provider changelog hints that https://github.com/apache/airflow/pull/35820/ is potentially the fix for the failure with our connection setup in the CI that was released in 8.5.0.

Therefore, I propose using azure provider>=8.5.0. However, the Airflow [constraints file](https://raw.githubusercontent.com/apache/airflow/constraints-2.8.0/constraints-3.8.txt) for version 2.8 specifies azure provider==8.4.0, which conflicts with this requirement. To address this, I am making changes to the pre-install script in our CI to install azure provider>=8.5.0 without relying on the constraints file, citing the reasons above.

closes: #1304

### CODE REVIEW

**Review by pankajastro — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -70,7 +70,7 @@ amazon = [
     "apache-airflow-providers-amazon[s3fs]>=3.0.0",
 ]
 google = ["apache-airflow-providers-google>=10.17.0"]
-microsoft = ["apache-airflow-providers-microsoft-azure"]
+microsoft = ["apache-airflow-providers-microsoft-azure>=8.5.0"]
```
Comment (tatiana) [inline pyproject.toml:73]: @pankajkoti would it make sense to try to fix the Airflow constraints files as well?

**Review by tatiana — APPROVED** [review]
Thank you very much, @pankajkoti , it's unexpected that the Airflow constraints files are not meeting our needs in this. I left a minor comment.

File: pyproject.toml (hunk 1 above)
Comment (pankajkoti) [reply pyproject.toml:73]: I believe those constraint files are generated at the time of release and aren't mean to be mutated in order to generate reproducible builds with those constraints. The constraint files ensure that Airflow tests (existing at that time) run fine with those dependencies and the docs mention that we could upgrade/downgrade dependencies as needed. I am afraid we can fix the constraints files at this point of time.

Ref: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#reproducible-airflow-installation

Comment (pankajkoti) [reply pyproject.toml:73]: any thoughts/ideas on this @pankajastro based on your previous experience working with constraints files?

Comment (tatiana) [reply pyproject.toml:73]: Thanks a lot for the explanation, @pankajkoti, it makes sense!

It's a shame - I initially hoped we could leverage the Airflow constraints as they are to solve Cosmos dependencies - but I did realise this didn't quite work on a previous PR.

**Pull request merged**